qa/tasks/fwd_scrub.py

```diff
@@ -0,0 +1,230 @@
+        try:
+            while not self.stopping.is_set():
+                ranks = self.fs.get_all_mds_rank()
+
+                for r in ranks:
+                    scrubber = MDSRankScrubber(self.fs, r, self.scrub_timeout)
+                    self.lock.acquire()
+                    try:
+                        self.scrubbers.append(scrubber)
+                    finally:
+                        self.lock.release()
+                    scrubber.start()
+
+                # wait for all scrubbers to complete
+                self.log("joining all scrubbers")
+                joinall(self.scrubbers)
+
+                for s in self.scrubbers:
+                    if s.exception is not None:
+                        raise RuntimeError('error during scrub thrashing')
+
+                self.lock.acquire()
+                try:
+                    self.scrubbers.clear()
+                finally:
+                    self.lock.release()
+
+                sleep(self.sleep_between_iterations)
+        except GreenletExit:
+            pass
+
+
+def stop_all_fwd_scrubbers(thrashers):
+    for thrasher in thrashers:
+        if not isinstance(thrasher, ForwardScrubber):
+            continue
+        thrasher.stop()
+        if thrasher.exception is not None:
+            raise RuntimeError(f"error during scrub thrashing: {thrasher.exception}")
+        thrasher.join()
+
+
+@contextlib.contextmanager
+def task(ctx, config):
+    """
+    Stress test the mds by running scrub iterations while another task/workunit
+    is running.
+    Example config:
+
+    - fwd_scrub:
+      scrub_timeout: 300
+      sleep_between_iterations: 1
+    """
+
+    mds_cluster = MDSCluster(ctx)
+
+    if config is None:
+        config = {}
+    assert isinstance(config, dict), \
+        'fwd_scrub task only accepts a dict for configuration'
+    mdslist = list(teuthology.all_roles_of_type(ctx.cluster, 'mds'))
+    assert len(mdslist) > 0, \
+        'fwd_scrub task requires at least 1 metadata server'
+
+    (first,) = ctx.cluster.only(f'mds.{mdslist[0]}').remotes.keys()
+    manager = ceph_manager.CephManager(
+        first, ctx=ctx, logger=log.getChild('ceph_manager'),
+    )
+
+    # make sure everyone is in active, standby, or standby-replay
+    log.info('Wait for all MDSs to reach steady state...')
+    status = mds_cluster.status()
+    while True:
+        steady = True
+        for info in status.get_all():
+            state = info['state']
+            if state not in ('up:active', 'up:standby', 'up:standby-replay'):
+                steady = False
+                break
+        if steady:
+            break
+        sleep(2)
+        status = mds_cluster.status()
+
+    log.info('Ready to start scrub thrashing')
+
+    manager.wait_for_clean()
+    assert manager.is_clean()
+
+    if 'cluster' not in config:
+        config['cluster'] = 'ceph'
+
+    for fs in status.get_filesystems():
+        fwd_scrubber = ForwardScrubber(Filesystem(ctx, fs['id']),
```

Comment:
````
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/run_tasks.py", line 94, in run_tasks
    manager.__enter__()
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20201213.040840/qa/tasks/fwd_scrub.py", line 218, in task
    fwd_scrubber = ForwardScrubber(Filesystem(ctx, fs['id']),
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20201213.040840/qa/tasks/cephfs/filesystem.py", line 465, in __init__
    self.ec_profile = fs_config.get('cephfs_ec_profile')
AttributeError: 'int' object has no attribute 'get'
2020-12-13T06:36:50.780 ERROR:teuthology.run_tasks: Sentry event: https://sentry.ceph.com/organizations/ceph/?query=31581b2e11264606ac81c2876f0d63a1
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/run_tasks.py", line 94, in run_tasks
    manager.__enter__()
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20201213.040840/qa/tasks/fwd_scrub.py", line 218, in task
    fwd_scrubber = ForwardScrubber(Filesystem(ctx, fs['id']),
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20201213.040840/qa/tasks/cephfs/filesystem.py", line 465, in __init__
    self.ec_profile = fs_config.get('cephfs_ec_profile')
AttributeError: 'int' object has no attribute 'get'
````

From: /ceph/teuthology-archive/pdonnell-2020-12-13_06:09:39-fs-wip-pdonnell-testing-20201213.040840-distro-basic-smithi/5704874/teuthology.log

Reply:
needed rebase and fix due to Ramana's changes
testing is in progress

Reply:
[testing is complete](http://pulpito.front.sepia.ceph.com/mchangir-2020-12-13_23:21:06-fs:workload-wip-mchangir-qa-forward-scrub-task-distro-basic-smithi/5705548/)
PR is good for review

Reply:
@mchangir please run with `-s fs:workload --filter scrub/yes --limit 20`

Reply:
[`--limit 20` teuthology job run](http://pulpito.front.sepia.ceph.com/mchangir-2020-12-14_17:09:38-fs:workload-wip-mchangir-qa-forward-scrub-task-distro-basic-smithi/)

Reply:
Great, thanks.